File: 0198-house-robber/0198-house-robber.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var rob = function(nums) {

    let maxSum = new Array(nums+1)
    maxSum[0] = 0
    maxSum[1] = nums[0]
    
    for(let i=1;i<nums.length; i++) {
        maxSum[i+1] = Math.max(maxSum[i], maxSum[i-1] + nums[i])
    }
    
    return maxSum[nums.length]
};

File: 0198-house-robber/0198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        int n = nums.length;
        
        if(n == 0)
            return 0;
        
        int dp[] = new int[n + 1];
        
        // we can gain 0 by robbing 0 houses
        dp[0] = 0;
         
        // nums[0] can be gained by robbing 1st house
        dp[1] = nums[0];
        
        for(int i=1; i<n; i++) {
            dp[i+1] = Math.max(dp[i], dp[i-1] + nums[i]);
        }
        
        return dp[n];
    }
}